lib/charms/mysql/v0/mysql.py

```diff
@@ -610,21 +622,24 @@ def set_secret(self, scope: Scopes, key: str, value: Optional[str]) -> None:
             return self.remove_secret(scope, key)
 
         peers = self.model.get_relation(PEER)
-        if scope == APP_SCOPE:
-            self.peer_relation_app.update_relation_data(peers.id, {key: value})
-        elif scope == UNIT_SCOPE:
-            self.peer_relation_unit.update_relation_data(peers.id, {key: value})
+
+        fallback_key_to_secret_key = {v: k for k, v in SECRET_KEY_FALLBACKS.items()}
+        if key in fallback_key_to_secret_key:
```

Comment:
this would be true when an old charm (running on juju <= 3.1.4 when secrets were not introduced) would upgrade to the new charm. in this case, the value from the old fallback_key should be deleted, and new value should be stored in juju secrets with the new key

this functionality was present in the charm, but was accidentally removed with a refactor

Reply:
i believe that the old (pre backups) code called `set_secret` with keys like (root_password) -> this is incompatible with juju secrets due to the underscore. in the this old code, `set_secret` set password in peer databag

the logic is as follows:
- if the old/fallback key exists in either the databag (or secret - but cannot exist in secret), then delete it
- update the new key in either the secret or the databag using data_interfaces